spec/models/ems_refresh/parsers/kubernetes_spec.rb

```diff
@@ -62,4 +62,29 @@
       end
     end
   end
+
+  describe "parse_container_image" do
+    shared_image_without_host = "shared/image"
+    shared_image_with_host = "host:1234/shared/image"
+    shared_ref = "shared:ref"
+    unique_ref = "unique:ref"
+
+    it "returns unique object for same image but different ref/id" do
+      [shared_image_with_host, shared_image_without_host].each do |shared_image|
+        first_obj  = parser.parse_container_image(shared_image, shared_ref)
+        second_obj = parser.parse_container_image(shared_image, unique_ref)
+
+        first_obj.should_not be(second_obj)
```

Comment:
That will cover cases where the data_index.store_path() fails. Is it an acceptable scenario?
If it is not acceptable, I guess we should fail the test.

Ideally both scenarios need testing, each with its own expected results. But for that, we may need to mock the data_index.store_path() or cause the original one to fail.

Would you still recommend passing the single test for both scenarios?